net/net.go

```diff
@@ -41,14 +57,37 @@ func RemoteHost(r *http.Request) net.IP {
 	return parse(r.RemoteAddr)
 }
 
-// RemoteHostFromLast returns the remote address of the client. When
+// RemoteAddrFromLast returns the remote address of the client. When
 // the 'X-Forwarded-For' header is set, then it is used instead. This
 // is known to be true for AWS Application LoadBalancer. AWS docs
 // https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html
 //
 // Example:
 //
 //	X-Forwarded-For: ip-address-1, ip-address-2, client-ip-address
+func RemoteAddrFromLast(r *http.Request) netip.Addr {
+	ffs := r.Header.Get("X-Forwarded-For")
+	if ffs == "" {
+		addr, _ := netip.ParseAddr(stripPort(r.RemoteAddr))
+		return addr
+	}
+
+	if i := strings.LastIndex(ffs, ","); i != -1 {
+		last := ffs[i+1:]
+		if addr, err := netip.ParseAddr(strings.TrimSpace(last)); err == nil {
+			return addr
+		}
+	}
+
+	addr, err := netip.ParseAddr(strings.TrimSpace(ffs))
```

Comment:
If netip.ParseAddr on line 77 returns error we attempt to parse again at line 82.
I think it should be like
```go
    last := ffs
    if i := strings.LastIndex(ffs, ","); i != -1 {
        last := ffs[i+1:]
    }
    addr, err := netip.ParseAddr(strings.TrimSpace(last))
    if err != nil {
        addr, _ := netip.ParseAddr(stripPort(r.RemoteAddr))
        return addr
    }
    return addr
```

Reply:
I don't think it's different but mine is more readable 🙂

Reply:
fixed